frontend/src/component/ui/ChangeLogModal.tsx

```diff
@@ -25,6 +26,20 @@ export class ChangeLogModal extends Component<Props, State> {
     shouldShowFullChangeLog: false
   };
 
+  private modalRef = React.createRef<Modal>();
+
+  componentDidUpdate(prevProps: Props) {
+    if (this.props.shouldShowModal === prevProps.shouldShowModal) {
+      return;
+    }
+
+    if (!this.props.shouldShowModal) {
+      return;
+    }
```

Comment:
When to open or close a modal, should be decoupled from the modal's implementation. Currently, you are the creator of this component. However, other engineers may want to extend the behavior of your modal, for example, also check whether whether an user is logged in. The only place they need to for should be Home.tsx. This reduces the efforts they need to make changes. If everyone on the project follows those principals, it's also easier and faster for you to add new features. This is called **embrace changes**.  You can read more software design principles we use at here: http://s/solid 😄

Reply:
Suggestion: remove line 32 to line 40.